sonic-thermalctld/scripts/thermalctld

```diff
@@ -640,6 +640,36 @@ class TemperatureUpdater(logger.Logger):
                                         )
         warning = warning | temperature_status.under_temperature
 
+        THRESH_NAME_LIST = ['high_threshold', 'low_threshold', 'critical_high_threshold', 'critical_low_threshold']
+        THRESH_COMP_VAL_LIST = [high_threshold, low_threshold, high_critical_threshold, low_critical_threshold]
+        THRESH_FUNC_LIST = [thermal.set_high_threshold, thermal.set_low_threshold, thermal.set_high_critical_threshold, thermal.set_low_critical_threshold]
+
+        ovalues = self.table.get(name)
+
+        for first, second in ovalues[1]:
+            if second == NOT_AVAILABLE:
+                continue
+
+            th_type = 0
+            if first in THRESH_NAME_LIST:
+                th_type = THRESH_NAME_LIST.index(first)
+            else:
+                continue
+
+            if second != str(THRESH_COMP_VAL_LIST[th_type]):
```

Comment:
"second" here is the value of a threshold, while THRESH_COMP_VAL_LIST[th_type] is a element in THRESH_COMP_VAL_LIST, i suppose they never equal.

Reply:
Second is new value in string and THRESH_COMP_VAL_LIST[th_type] is current value in float, which I cast to string, so they should be equal.
```
second: <class 'str'>111.0
THRESH_COMP_VAL_LIST[th_type]: <class 'float'>111.0
```